---
title: "Two Sample RNA-seq sample comparison"
author: "adowneywall"
date: "7/9/2020"
output: github_document
always_allow_html: true
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r include=FALSE}
library(ggplot2, quietly = TRUE)
library(cowplot, quietly = TRUE)
library(dplyr, quietly = TRUE)
library(kableExtra, quietly = TRUE)
library(corrplot)
```
  
## STAR Mapping Comparison

```{r echo=FALSE}
#### STAR Data
setwd("~/Github/updatedOysterTranscriptomeMappingComparison/")
## STAR Mapping Comparison
colNames <- c("Sample_Unique","Sample","Genome","Parameters",
              "InputReads","UniqueReads","UniqueReadPercent",
              "totalSplices","MisMatchRate",
              "MultiMapReads","MultiMapReadPercent",
              "UnMappedMismatchPercent","UnMappedShortPercent")

readInSTARSummary <- function(folder=NULL,genome="Hap",param="custom",col.names=NULL){
  fl <- list.files(folder,pattern = "final.out")
  y <- NULL
  for(i in 1:length(fl)){
    x <- read.delim(paste0(folder,fl[i]),header = FALSE,stringsAsFactors = FALSE)
    x_sub <- x[c(5,8,9,11,17,23,24,28,29),2]
    ifelse(genome == "Hap",
           ID <- substr(fl[i],1,5),
           ID <- substr(fl[i],4,8))
    ID_U <- paste0(ID,"_",genome,"_",param)
    y <- rbind(y,c(ID_U,ID,genome,param,x_sub))
  }
  y <- data.frame(y,stringsAsFactors = FALSE)
  colnames(y) <- col.names
  for(i in 1:nrow(y)){y[i,c(7,9,11,12,13)] <- substr(y[i,c(7,9,11,12,13)],1,4)}
  return(y)
}

hap_default_STAR <- readInSTARSummary("data/samples/haploTigGenome_defaultParameters/",
                  "Hap","default",colNames)
hap_custom_STAR <- readInSTARSummary("data/samples/haploTigGenome_customParameters/",
                                      "Hap","custom",colNames)
og_custom_STAR <- readInSTARSummary("data/samples/originalGenome_customParameters/",
                                      "original","custom",colNames)
df_STAR <- rbind(hap_default_STAR,hap_custom_STAR,og_custom_STAR)

sub <- c(which(df_STAR$Sample == c("17005")),which(df_STAR$Sample == c("17099")))
df_overlap <- df_STAR[sub,]
```

```{r echo=FALSE}
#### Table
 kable(df_overlap) %>%
  kable_styling() %>%
  scroll_box(width = "800px", height = "250px")
```


```{r  echo=FALSE, fig.width=10,fig.height=8}
#### Plots
p1 <- ggplot(df_overlap,aes(x=Sample,
                            y=as.numeric(UniqueReadPercent),
                            colour=interaction(Genome,Parameters))) + 
  geom_jitter(size=5) +
  labs(colour="Genome.Parameter",y="Unique Map %") +
  theme_cowplot()
p2 <- ggplot(df_overlap,aes(x=Sample,
                   y=as.numeric(MultiMapReadPercent),
                   colour=interaction(Genome,Parameters))) + 
  geom_jitter(size=5) +
  labs(colour="Genome.Parameter",y="Multi Map %") +
  theme_cowplot()
p3 <- ggplot(df_overlap,aes(x=Sample,
                            y=as.numeric(MisMatchRate),
                            colour=interaction(Genome,Parameters))) + 
  geom_jitter(size=5) +
  labs(colour="Genome.Parameter",y="Mismatch Rate (%)") +
  theme_cowplot()
p4 <- ggplot(df_overlap,aes(x=Sample,
                            y=as.numeric(UnMappedShortPercent),
                            colour=interaction(Genome,Parameters))) + 
  geom_jitter(size=5) +
  labs(colour="Genome.Parameter",y="Unmapped Read %") +
  theme_cowplot()

plot_grid(p1,p2,p3,p4,labels = c("A","B","C","D"),nrow=2)
```

## RSEM Comparison (sample 17005)

```{r}
setwd("~/Github/updatedOysterTranscriptomeMappingComparison/")
OG_RSEM <- read.delim("data/samples/originalGenome_customParameters/17005_.genes.results.txt")
Hap_RSEM <- read.delim("data/samples/haploTigGenome_customParameters/17005.genes.results.txt")

OG_RSEM_17099 <- read.delim("data/samples/originalGenome_customParameters/17099_.genes.results.txt")
HAP_RSEM_17099 <- read.delim("data/samples/haploTigGenome_customParameters/17099.genes.results")
# TPM
tpmMat <- data.frame(OG_17005=OG_RSEM$TPM,Hap_17005=Hap_RSEM$TPM,
                     OG_17099=OG_RSEM_17099$TPM,Hap_17099=HAP_RSEM_17099$TPM)
# Length
lengthMat <- data.frame(OG_17005=OG_RSEM$length,Hap_17005=Hap_RSEM$length,
                        OG_17099=OG_RSEM_17099$length,Hap_17099=HAP_RSEM_17099$length)
# Expected Count
ExpCountMat <- data.frame(OG_17005=OG_RSEM$expected_count,Hap_17005=Hap_RSEM$expected_count,
                          OG_17099=OG_RSEM_17099$expected_count,Hap_17099=HAP_RSEM_17099$expected_count)

```

### CountSummary
```{r}
#17005 - Number of genes with at least 1 TPM - original genome
sum(tpmMat$OG_17005 > 1)
#17005- Number of genes with at least 1 TPM - reduced genome
sum(tpmMat$Hap_17005 > 1)
#17099 - Number of genes with at least 1 TPM - original genome
sum(tpmMat$OG_17099 > 1)
#17099- Number of genes with at least 1 TPM - reduced genome
sum(tpmMat$Hap_17099 > 1)
```

### Correlations

**TPM (Transcripts Per Million)**
```{r}
corrplot.mixed(cor(tpmMat))
```

**Gene Length**
```{r}
corrplot.mixed(cor(lengthMat))
```

**Expected Count**
```{r}
corrplot.mixed(cor(ExpCountMat))
```

### TPM - Plot (Sample 17005, custom parameters, genome comparison)

```{r,fig.width=10,fig.height=15}
p1 <- ggplot(tpmMat,aes(x=OG_17005,y=Hap_17005)) + 
  geom_point() +
  labs(x="Original Genome (TPM)", y = "Reduced Genome (TPM)",title="All Loci") +
  theme_cowplot()

p2 <- ggplot(tpmMat,aes(x=OG_17005,y=Hap_17005)) + 
  xlim(0,200) + ylim(0,200) +
  labs(x="Original Genome (TPM)", y = "Reduced Genome (TPM)",title="Max 200") +
  geom_point() +
  theme_cowplot()

p3 <- ggplot(tpmMat,aes(x=OG_17005,y=Hap_17005)) + 
  xlim(0,50) + ylim(0,50) +
  labs(x="Original Genome (TPM)", y = "Reduced Genome (TPM)",title="Max 50") +
  geom_point() +
  theme_cowplot()

plot_grid(p1,p2,p3,nrow = 3)
```

### TPM - Plot (Sample 17099, custom parameters, genome comparison)

```{r,fig.width=10,fig.height=15}
p1 <- ggplot(tpmMat,aes(x=OG_17099,y=Hap_17099)) + 
  geom_point() +
  labs(x="Original Genome (TPM)", y = "Reduced Genome (TPM)",title="All Loci") +
  theme_cowplot()

p2 <- ggplot(tpmMat,aes(x=OG_17099,y=Hap_17099)) + 
  xlim(0,200) + ylim(0,200) +
  labs(x="Original Genome (TPM)", y = "Reduced Genome (TPM)",title="Max 200") +
  geom_point() +
  theme_cowplot()

p3 <- ggplot(tpmMat,aes(x=OG_17099,y=Hap_17099)) + 
  xlim(0,50) + ylim(0,50) +
  labs(x="Original Genome (TPM)", y = "Reduced Genome (TPM)",title="Max 50") +
  geom_point() +
  theme_cowplot()

plot_grid(p1,p2,p3,nrow = 3)
```
  
## Comparing among samples  
  
To evaluate the impact of the genome mapping among samples I calculated the difference in coverage (TPM) between genome mapping approaches and examine whether this value difference among samples. The difference discussed below was calculated as:  

$$(X_{TPM,Hap})_{i}-(X_{TPM,OG})_{i}$$  
Where $X$ is the sample, and $i$ is the gene.  
  

### All Genes

**Absolute difference - All genes**
```{r,fig.width=10,fig.height=10}
plot(c(abs(tpmMat$Hap_17005-tpmMat$OG_17005))~c(abs(tpmMat$Hap_17099-tpmMat$OG_17099)),
     xlab="TPM difference 17099",ylab="TPM difference 17005")
abline(a=0,b = 1,col="red")
```

**Absolute difference - coverage < 500**
```{r,fig.width=10,fig.height=10}
plot(c(abs(tpmMat$Hap_17005-tpmMat$OG_17005))~c(abs(tpmMat$Hap_17099-tpmMat$OG_17099)),
     xlab="TPM difference 17099",ylab="TPM difference 17005",
     xlim=c(0,200),ylim=c(0,200))
abline(a=0,b = 1,col="red")
```

**Absolute difference - coverage < 50**
```{r,fig.width=10,fig.height=10}
plot(c(abs(tpmMat$Hap_17005-tpmMat$OG_17005))~c(abs(tpmMat$Hap_17099-tpmMat$OG_17099)),
     xlab="TPM difference 17099",ylab="TPM difference 17005",
     xlim=c(0,50),ylim=c(0,50))
abline(a=0,b = 1,col="red")
```

**Proportion difference (adjusted by max TPM)**
```{r,fig.width=10,fig.height=10}
library(matrixStats)
hapProp_17005 <- c(tpmMat$Hap_17005-tpmMat$OG_17005)/rowMaxs(as.matrix(tpmMat[,1:2]))
hapProp_17099 <- c(tpmMat$Hap_17099-tpmMat$OG_17099)/rowMaxs(as.matrix(tpmMat[,3:4]))

plot(hapProp_17005~hapProp_17099,
     xlab="Prop TPM difference 17099",ylab="Prop TPM difference 17005")
abline(a=0,b=1,col="red",lwd=2)
```

### Filter dataset
Looking at genes with moderate coverage (diff TPM >= 1 in at least 1 sample/mapping)

```{r}
#Filter gene with no coverage in either samples
minCov_all <- which(rowMins(as.matrix(tpmMat)) >= 1)
tpmMat_filt <- tpmMat[minCov_all,]

diff_17005 <- abs(tpmMat_filt$Hap_17005 - tpmMat_filt$OG_17005)
diff_17099 <- abs(tpmMat_filt$Hap_17099 - tpmMat_filt$OG_17099)
```

**Absolute difference - All genes**
```{r,fig.width=10,fig.height=10}
plot(diff_17005~diff_17099,
     xlab="TPM difference 17099",ylab="TPM difference 17005")
abline(a=0,b=1,col="red")
```

**Absolute difference - coverage < 100**
```{r,fig.width=10,fig.height=10}
plot(diff_17005~diff_17099,xlim=c(0,100),ylim=c(0,100),
     xlab="TPM difference 17099",ylab="TPM difference 17005")
abline(a=0,b=1,col="red")
```

**Absolute difference - All genes (log transformed)**
```{r,fig.width=10,fig.height=10}
plot(log(diff_17005)~log(diff_17099),
     xlab="TPM difference 17099 (log)",ylab="TPM difference 17005 (log)")
abline(a=0,b=1,col="red")
```

**Number of genes with log-tpm fold > 2 among samples**
```{r}
## Sum 
sum(abs(log(diff_17005)-log(diff_17099))>2)
```







